lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -79,6 +79,12 @@ def replace_duplicates_with_catalog_resources(generated)
 
   def contain_generated_resources_in(resource, made)
     sentinel = Puppet::Type.type(:whit).new(:name => "completed_#{resource.title}", :catalog => resource.catalog)
+    # Tag the completed whit with all of the tags of the generating resource
+    # except the type name to allow event propogation to succeed beyond the whit
+    # "boundary" when filtering resources with tags. All resources are tagged
+    # with their type name by default, so we don't include this tag to avoid any
+    # confusion from the whit containing tags for two resource types.
+    sentinel.tag(*resource.tags.delete(resource.type.to_s))
```

Comment:
So changing to `resource.parameters[:tag].value` makes the new tests in `additional_resource_generator_spec.rb` fail... so that probably answers my question.